### PR TITLE
[dtensor] add debug tool to track op coverage

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100124

This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

### CODE REVIEW

File: torch/distributed/_tensor/debug/op_coverage.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def fwd_bwd_compiler(fx_g, _):
+    graphs.append(fx_g)
+    return fx_g
+
+
+def get_inductor_decomp_graphs(model: nn.Module, args, kwargs):
+    """
+    Convinient util to get the fwd and bwd graphs of an arbitrary model
```
Comment (XilunWu) [inline torch/distributed/_tensor/debug/op_coverage.py:19]: typo: convinient -> convenient

File: test/distributed/_tensor/debug/test_op_coverage.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class TestOpCoverage(TestCase):
+    def test_trace_with_inductor_decomp(self):
+        model = SimpleMLP()
+        args = (torch.randn(8, 50),)
+        kwargs = {}
+        graphs = get_inductor_decomp_graphs(model, args, kwargs)
+        assert len(graphs) == 2, "Expect fwd + bwd graphs"
+        self.assertIsInstance(graphs[0], torch.fx.GraphModule)
```
Comment (XilunWu) [inline test/distributed/_tensor/debug/test_op_coverage.py:29]: question: will we check the content of the graphs in this test or another? Does this PR include checking the graph content or just validating that the util can return 1 fwd and 1 bwd graphs.

**Review by XilunWu — APPROVED** [review]
LGTM! a good start for op coverage!!

File: test/distributed/_tensor/debug/test_op_coverage.py (hunk 2 above)
Comment (wanchaol) [reply test/distributed/_tensor/debug/test_op_coverage.py:29]: This PR would not check the content of the graph in the tests, I added a `print_op_summary` function that would print the op coverage summary

**Pull request closed without merging**